### PR TITLE
Bump min version

### PR DESCRIPTION
**Pull request by jburel**
* Drop Ice 3.5
* Drop Java 1.7
* Drop CentOS 6

I have **not** changed the version-requirements page in this PR.

The files under ``omero/sysadmins/unix/walkthrough/`` will be automatically replaced when ``autogen`` is run

### CODE REVIEW

File: omero/conf.py [hunk 1]
```diff
@@ -66,9 +66,9 @@
 .. |previousversion| replace:: %s
 .. |current_dbver|  replace:: %s
 .. |previous_dbver|  replace:: %s
-.. |iceversion| replace:: 3.5.1
+.. |iceversion| replace:: 3.6.3
```
Comment (mtbc) [inline omero/conf.py]: not 3.6.4?

File: omero/developers/Cpp.rst [hunk 2]
```diff
@@ -31,10 +31,6 @@ Prerequisites
     limitation in the Windows PE-COFF executable format used for DLLs,
     even on 64-bit systems.  Workarounds include:
```
Comment (mtbc) [inline omero/developers/Cpp.rst]: Maybe omit "Workaround" and bullet and just start "It is hoped ..."

**Comment by jburel** [comment]
@mtbc updated

File: omero/developers/Cpp.rst [hunk 3]
```diff
@@ -29,10 +29,9 @@ Prerequisites
     LNK1189: library limit of 65535 objects exceeded`` and will be
     unable to build |OmeroCpp| for Windows as a result of a 16-bit
     limitation in the Windows PE-COFF executable format used for DLLs,
-    even on 64-bit systems.  Workarounds include:
-
-    - it is hoped that :program:`Ice` 3.7 (when released) will resolve
-      the problem since it generates far fewer symbols than 3.6
+    even on 64-bit systems.  It is hoped that :program:`Ice` 3.7
+    (when released) will resolve the problem since it generates
```
Comment (mtbc) [inline omero/developers/Cpp.rst]: It already is released?

**Comment by mtbc** [comment]
Great, thank you @jburel.

**Comment by mtbc** [comment]
Maybe also something for the last bit of console script on https://ci.openmicroscopy.org/job/OMERO-DEV-merge-docs/ws/src/omero/_build/html/users/cli/installation.html.

**Comment by mtbc** [comment]
Figure also CI contrib docs can be updated outwith this PR after we have CI sorted properly for 5.5.0.

**Review by mtbc — APPROVED** [review]

**Pull request merged**